Microsoft.VCRTForwarders.140.targets

```diff
@@ -4,34 +4,91 @@
-  </ItemGroup>
+        var properties = new Dictionary<string, string>
+        {
+          { "Configuration", Configuration },
+          { "Platform", Platform }
+        };
+        
+        // Per MSDN, _DEBUG define can be checked to determine if debug CRT is in use.
+        var collection = new ProjectCollection(properties);
+        var project = collection.LoadProject(ProjectFile);
+        ProjectMetadata def = project.AllEvaluatedItemDefinitionMetadata
+                                     .LastOrDefault(p => p.ItemType == "ClCompile" && p.Name == "PreprocessorDefinitions");
+        Boolean useDebug = def != null && (";" + def.EvaluatedValue + ";").Contains(";_DEBUG;");
 
-  <!-- arm64 -->
-  <ItemGroup Condition="'$(Configuration)' == 'Debug' And '$(Platform)' == 'ARM64'">
-    <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\..\runtimes\win10-arm64\native\debug\*.dll" />
-    <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\..\runtimes\win10-arm64\native\release\*.dll" />
-  </ItemGroup>
-   <ItemGroup Condition="'$(Configuration)' == 'Release' And '$(Platform)' == 'ARM64'">
-    <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\..\runtimes\win10-arm64\native\release\*.dll" />
-  </ItemGroup>
+        // There seem to be cases where even if _DEBUG is not found, debug CRT is used based on RuntimeLibrary.
+        if(!useDebug)
+        {
+          ProjectMetadata runtimeLibrary = project.AllEvaluatedItemDefinitionMetadata
+                                                  .LastOrDefault(p => p.ItemType == "ClCompile" && p.Name == "RuntimeLibrary");
+          useDebug = runtimeLibrary != null && (String.Compare(runtimeLibrary.EvaluatedValue, "MultiThreadedDebugDLL", true) == 0 ||
+                                                String.Compare(runtimeLibrary.EvaluatedValue, "MultiThreadedDebug", true) == 0);
+        }
+
+        TaskOutput = useDebug;
+        
+        ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+  
+  <Target Name="ResolveVCRTForwarderReferences">
+    <UseDebugCRT ProjectFile="%(ProjectReferenceWithConfiguration.Identity)"
+                 Configuration="%(ProjectReferenceWithConfiguration.Configuration)"
+                 Platform="%(ProjectReferenceWithConfiguration.Platform)"
+                 Condition="'@(ProjectReferenceWithConfiguration)' != '' And '$(VCRTForwarders-IncludeDebugCRT)' == ''">
+      <Output ItemName="TaskOutput" TaskParameter="TaskOutput"/>
+    </UseDebugCRT>
+    <PropertyGroup>
+      <VCRTForwarders-IncludeDebugCRT Condition="'%(TaskOutput.Identity)' == 'true'">true</VCRTForwarders-IncludeDebugCRT>
+    </PropertyGroup>
+
+    <!-- x86 -->
+    <ItemGroup Condition="$(VCRTForwarders-IncludeDebugCRT) == true And ('$(Platform)' == 'Win32' Or '$(Platform)' == 'x86')">
+      <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\..\runtimes\win10-x86\native\debug\*.dll" />
```

Comment:
It seems like you should be able to encapsulate the win32 / x86 logic in one place to set up a VCRT platform variable, and a native / debug based on IncludeDebugCRT, then reduce this section to be not architecture-specific, and instead just use the two variables...

Reply:
<PropertyGroup>
<VCRTPlatform>$(Platform)</>
<VCRTPlatform Condition="'$(Platform)=='Win32'")>x86</>
</>

---
In reply to: [409810456](https://github.com/microsoft/vcrt-forwarders/pull/20#discussion_r409810456) [](ancestors = 409810456)

Reply:
I agree, working on this change to simplify.

Reply:
Addressed.